res/controllers/common-controller-scripts.js

```diff
@@ -158,511 +487,205 @@ script.LIBRARY_COLUMNS = Object.freeze({
-            + ((inputValue - this.midInput) / (this.maxInput - this.midInput))
-            * (this.maxOutput - this.midOutput);
-    }
-    if (this.softMode) {
-        const currentValue = engine.getValue(group, this.mappedFunction);
-        let currentRelative = 0.0;
-        if (currentValue <= this.midOutput) {
-            currentRelative = this.minInput
-                + ((currentValue - this.minOutput) / (this.midOutput - this.minOutput))
-                * (this.midInput - this.minInput);
+// @ts-ignore Same identifier for class and instance needed for backward compatibility
+class Control {
+    constructor(mappedFunction, softMode) {
+        // These defaults are for MIDI controllers
+        this.minInput = 0;
+        this.midInput = 0x3F;
+        this.maxInput = 0x7F;
+        // ----
+        this.minOutput = -1.0;
+        this.midOutput = 0.0;
+        this.maxOutput = 1.0;
+        this.mappedFunction = mappedFunction;
+        this.softMode = softMode;
+        this.maxJump = 10;
+    }
+    setValue(group, inputValue) {
+        let outputValue = 0;
+        if (inputValue <= this.midInput) {
+            outputValue = this.minOutput
+                + ((inputValue - this.minInput) / (this.midInput - this.minInput))
+                * (this.midOutput - this.minOutput);
         } else {
-            currentRelative = this.midInput
-                + ((currentValue - this.midOutput) / (this.maxOutput - this.midOutput))
-                * (this.maxInput - this.midInput);
+            outputValue = this.midOutput
+                + ((inputValue - this.midInput) / (this.maxInput - this.midInput))
+                * (this.maxOutput - this.midOutput);
         }
-        if (inputValue > currentRelative - this.maxJump
+        if (this.softMode) {
+            const currentValue = engine.getValue(group, this.mappedFunction);
+            let currentRelative = 0.0;
+            if (currentValue <= this.midOutput) {
+                currentRelative = this.minInput
+                    + ((currentValue - this.minOutput) / (this.midOutput - this.minOutput))
+                    * (this.midInput - this.minInput);
+            } else {
+                currentRelative = this.midInput
+                    + ((currentValue - this.midOutput) / (this.maxOutput - this.midOutput))
+                    * (this.maxInput - this.midInput);
+            }
+            if (inputValue > currentRelative - this.maxJump
                 && inputValue < currentRelative + this.maxJump) {
+                engine.setValue(group, this.mappedFunction, outputValue);
+            }
+        } else {
             engine.setValue(group, this.mappedFunction, outputValue);
         }
-    } else {
-        engine.setValue(group, this.mappedFunction, outputValue);
     }
-};
+}
+// Add class Control to the Global JavaScript object
+// @ts-ignore Same identifier for class and instance needed for backward compatibility
+this.Control = Control;
```

Comment:
I don't know if `this` in the global context works. I think `globalThis` is standardized.

Reply:
`this` works here and refers the `globalThis`.

Reply:
good to know, that assumption probably won't hold for ES modules though. but we can revisit that topic when esm-based mappings are a thing.